service/src/main/java/uk/nhs/adaptors/gp2gp/common/storageconnectors/AzureStorageConnector.java

```diff
@@ -0,0 +1,39 @@
+package uk.nhs.adaptors.gp2gp.common.storageconnectors;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import org.springframework.beans.factory.annotation.Autowired;
+import com.azure.storage.blob.BlobClient;
+import com.azure.storage.blob.BlobContainerClient;
+
+public class AzureStorageConnector implements StorageConnector {
+
+    @Autowired
+    private BlobContainerClient containerClient;
+
+    protected AzureStorageConnector() {}
+
+    @Override
+    public void uploadToStorage(InputStream is, String filename) throws StorageConnectorException {
+        try {
+            BlobClient blobClient = containerClient.getBlobClient(filename);
+            blobClient.upload(is, is.available());
+        } catch (IOException ioException) {
+            throw new StorageConnectorException("Error occurred uploading to Azure Storage", ioException);
+        }
+    }
+
+    @Override
+    public OutputStream downloadFromStorage(String filename) throws StorageConnectorException {
+        try {
+            BlobClient blobClient = containerClient.getBlobClient(filename);
+            OutputStream outputStream = new ByteArrayOutputStream();
+            blobClient.openInputStream().transferTo(outputStream);
```

Comment:
Creating a ByteArrayOutputStream and transferring blobClient.openInputStream() into it means we're loading the entire file into memory. What happens if the file is 1GB? What happens if the adaptor is processing many of these 1GB files at the same time?

From the story test and tech notes:

> The adaptor will not store any files locally and should avoid loading entire files into memory

Just return the OutputStream from blobClient.openInputStream() ;)